soroban-sdk/src/vec.rs

```diff
@@ -380,21 +380,26 @@ where
     }
 
     #[inline(always)]
-    pub fn get(&self, i: u32) -> Option<Result<T, T::Error>> {
+    pub fn get(&self, i: u32) -> Option<T> {
+        self.try_get(i).unwrap_optimized()
+    }
+
+    #[inline(always)]
+    pub fn try_get(&self, i: u32) -> Result<Option<T>, T::Error> {
         if i < self.len() {
-            let env = self.env();
-            let val = env.vec_get(self.obj, i.into()).unwrap_infallible();
-            Some(T::try_from_val(env, &val))
+            self.try_get_unchecked(i).map(|val| Some(val))
         } else {
-            None
+            Ok(None)
         }
     }
 
     #[inline(always)]
-    pub fn get_unchecked(&self, i: u32) -> Result<T, T::Error>
-    where
-        T::Error: Debug,
-    {
+    pub fn get_unchecked(&self, i: u32) -> T {
```

Comment:
I've kept the _unchecked variations. I think it's dubious to have four variations of a function, but the _unchecked variation seems pretty helpful when you know you're within bounds.